File: backend/app/services/blob_service.py
"""Azure Blob Storage 服务"""
from io import BytesIO
from typing import BinaryIO

from azure.identity import DefaultAzureCredential
from azure.storage.blob import BlobServiceClient

from ..config import get_settings


class BlobService:
    """Azure Blob 存储服务"""
    
    def __init__(self):
        settings = get_settings()
        self.container_name = settings.azure_storage_container_name
        credential = DefaultAzureCredential()
        self.client = BlobServiceClient(
            settings.azure_storage_account_url,
            credential=credential
        )

    def upload(self, file_name: str, file_data: bytes):
        """上传文件"""
        blob_client = self.client.get_blob_client(
            container=self.container_name,
            blob=file_name
        )
        blob_client.upload_blob(file_data, overwrite=True)

    def download(self, file_name: str) -> BinaryIO:
        """下载文件"""
        blob_client = self.client.get_blob_client(
            container=self.container_name,
            blob=file_name
        )
        download_stream = BytesIO()
        blob_client.download_blob().readinto(download_stream)
        download_stream.seek(0)
        return download_stream


# 单例实例
blob_service = BlobService()

File: Sync.Backend/Services/BlobService.cs
using Azure.Identity;
using Azure.Storage.Blobs;
using Microsoft.Extensions.Options;
using Sync.Backend.Models;

namespace Sync.Backend.Services;

/// <summary>Azure Blob storage service</summary>
public class BlobService
{
    private readonly BlobContainerClient _containerClient;

    public BlobService(IOptions<SyncSettings> options)
    {
        var settings = options.Value;
        var credential = new DefaultAzureCredential();
        var serviceClient = new BlobServiceClient(new Uri(settings.AzureStorageAccountUrl), credential);
        _containerClient = serviceClient.GetBlobContainerClient(settings.AzureStorageContainerName);
    }

    /// <summary>Upload a file (streaming)</summary>
    public async Task UploadAsync(string fileName, Stream data)
    {
        var blobClient = _containerClient.GetBlobClient(fileName);
        await blobClient.UploadAsync(data, overwrite: true);
    }

    /// <summary>Download a file (streaming)</summary>
    public async Task<Stream> DownloadAsync(string fileName)
    {
        var blobClient = _containerClient.GetBlobClient(fileName);
        var response = await blobClient.DownloadStreamingAsync();
        return response.Value.Content;
    }
}
